modules/market/src/main/java/com/opengamma/strata/market/curve/SeasonalNodalCurve.java

```diff
@@ -0,0 +1,638 @@
+   * With fixed initial value. Not a parameter.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final NodalCurve curve;
+  /**
+   * Describes the seasonal adjustments.
+   * The array has a dimension of 12, one element for each month.
+   * The adjustments are described as a perturbation to the existing values.
+   * No adjustment to the fixing value.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DoubleArray seasonality;
+  /**
+   * The shift type applied to the unadjusted value and the adjustment.
+   * (value, seasonality) -> adjustmentType.applyShift(value, seasonality).
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ShiftType adjustmentType;
+  
+  private final double xFixing;  // cached, not a property
+  private final double yFixing;  // cached, not a property
+  
+  /**
+   * Create an instance from the curve with initial point and total seasonal adjustment.
+   * 
+   * @param curve  the curve with initial fixing
+   * @param seasonality  the total seasonal adjustment for each month, with the first element the January adjustment
+   * @param adjustmentType  the adjustment type
+   * @return the seasonal curve instance
+   */
+  public static SeasonalNodalCurve of(
+      NodalCurve curve, 
+      DoubleArray seasonality, 
+      ShiftType adjustmentType) {
+    
+    return new SeasonalNodalCurve(curve, seasonality, adjustmentType);
+  }
+  
+  /**
+   * Create an instance from a curve without initial fixing point and month-on-month seasonal adjustment.
+   * <p>
+   * The total adjustment is computed by accumulation of the monthly adjustment, starting with no adjustment for the 
+   * last fixing month.
+   * 
+   * @param curveWithoutFixing  the curve without the fixing
+   * @param valuationDate  the valuation date of the curve
+   * @param lastMonth  the last month for which the fixing is known
+   * @param lastFixingValue  the value of the last fixing
+   * @param seasonalityDefinition  the seasonality definition, which is made of month-on-month adjustment 
+   * and the adjustment type
+   * @return the seasonal curve instance
+   */
+  public static SeasonalNodalCurve of(
+      NodalCurve curveWithoutFixing, 
+      LocalDate valuationDate,
+      YearMonth lastMonth,
+      double lastFixingValue,
+      SeasonalityDefinition seasonalityDefinition) {
+
+    ArgChecker.isTrue(lastMonth.isBefore(YearMonth.from(valuationDate)), 
+        "Last fixing month must be before valuation date");
+    double nbMonth = YearMonth.from(valuationDate).until(lastMonth, MONTHS);
+    DoubleArray x = curveWithoutFixing.getXValues();
+    ArgChecker.isTrue(nbMonth < x.get(0), "The first estimation month should be after the last known index fixing");
+    NodalCurve extendedCurve = curveWithoutFixing.withNode(nbMonth, lastFixingValue, ParameterMetadata.empty());
+    double[] seasonalityCompoundedArray = new double[12];
+    int lastMonthIndex = lastMonth.getMonth().getValue() - 2;
```

Comment:
This could be -1, is that correct? Maybe only subtract one and avoid adding one later.